gateway-client/src/lib.rs

```diff
@@ -0,0 +1,9 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2022 Oxide Computer Company
+
+//! Interface for API requests to a Management Gateway Service (MGS) instance
+
+omicron_common::generate_logging_api!("../openapi/gateway.json");
```

Comment:
The generated client's definition for `SpState` lost the "serial_number" field (which is present in `gateway.json`):
![Screenshot from 2022-03-09 09-47-03](https://user-images.githubusercontent.com/1435635/157465343-acb75b59-6221-48ba-ae9d-ea288e64084b.png)

Reply:
This was pretty interesting. Here's how we define `SpState`:

```rust
#[derive(Serialize, JsonSchema)]
#[serde(tag = "state")]
pub(crate) enum SpState {
    Disabled,
    Unresponsive,
    Enabled {
        serial_number: String,
        // TODO more stuff
    },
}
```

Note that it uses serde "internal" tagging i.e. where the output object for the `Enabled` variant will have a field "state" (set to a fixed value) is within the same object along with the field "serial_number". Here's how progenitor/typify interpret this which we can see with `cargo expand`:

```rust
    #[serde(tag = "state", content = "serial_number")]
    pub enum SpState {
        Disabled,
        Unresponsive,
        Enabled(String),
    }
```

Here we see "adjacent" tagging: progenitor (actually typify) has interpreted the JSON Schema in a different way that serializes identically. If and when we add the "more stuff", the object will no longer look like an adjacently tagged enum in which all variants have the same two fields (of which one must have constant values and the other is optional).